### PR TITLE
use golang 1.17 for build-root and add missing jobs

### PR DESCRIPTION
**Pull request by osherdp**
This PR addresses the following issues:
* We need to use the same golang version for the ``cloud_hotfix_releases`` branch, now that it points to the new code (see #26599). Without it it leads to [this kind of failure](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_assisted-service/3575/pull-ci-openshift-assisted-service-cloud_hotfix_releases-lint/1508516965038166016)
* We forgot to add ``unit-test`` and ``verify-generated-code`` jobs to this branch as well

/cc @filanov @gamli75 @eliorerz

### CODE REVIEW

**Comment by osherdp** [comment]
/hold
waiting for rehearse jobs

**Review by filanov — APPROVED** [review]

**Comment by osherdp** [comment]
/unhold
it does not rehearse the lint job, but it should do the trick
I will retest the failure job right after this gets merged

**Pull request merged**
